crossdock/crossdock_test.go

```diff
@@ -123,12 +113,45 @@ func TestCrossdock(t *testing.T) {
 	}
 }
 
+func wait(t *testing.T) {
+	attempts := 0
+	for {
+		if attempts > 9 {
+			t.Fatalf("could not talk to client in %d attempts", attempts)
+		}
+
+		attempts++
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
```

Comment:
This cancel isn't getting called at the end of this block. `defer` queues up the cancel for when the function exits (in reverse order, i.e., most recent first). That is probably not the behavior you were expecting.